.github/workflows/next-build-nightly.yml

```diff
@@ -0,0 +1,155 @@
+        name: Check public api
+      - run: yarn test:self
+        name: Self tests for tools
+      - run: yarn check:circular
+        name: Circular dependency Check
+      - run: yarn doc
+        name: API Doc Check
+      - uses: ./.github/actions/check-license
+        name: License Check
+      - if: failure()
+        name: Slack Notify
+        uses: rtCamp/action-slack-notify@v2.3.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+          SLACK_USERNAME: SDK Pipeline Bot
+          SLACK_TITLE: Build checks for v4 ${{ job.status }}
+          SLACK_MESSAGE: 'Test failed'
+          MSG_MINIMAL: event,actions url
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MSG_AUTHOR: ' '
+          SLACK_ICON: https://sap.github.io/cloud-sdk/img/logo.png
+  e2e-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+      - run: yarn install --frozen-lockfile --ignore-engines
+      - run: yarn test:e2e
+  canary-release-pre-check:
+    runs-on: ubuntu-latest
+    outputs:
+      skip: ${{ steps.date-check.outputs.skip-release }}
+    needs: [tests, checks, e2e-tests]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - id: date-check
+        name: Check if latest commit is within 24 hrs
+        run: |
+          lastCommitDate=$(git --no-pager log -n 1 ${{ github.ref_name }} --pretty=format:"%at")
+          curDate=$(date +%s)
+          dateDiff=$(expr $curDate - $lastCommitDate)
+          echo $lastCommitDate, $curDate, $dateDiff
+
+          if [[ $dateDiff -gt 86400 ]]
+          then
+              echo 'No new commit found on ${{ github.ref }} within the last 24 hrs.'
+              echo "skip-release=true" >> $GITHUB_OUTPUT
+          else
+              echo "skip-release=false" >> $GITHUB_OUTPUT
+          fi
+  canary-release:
+    if: ${{ needs.canary-release-pre-check.outputs.skip == 'false' }}
+    runs-on: ubuntu-latest
+    needs: [canary-release-pre-check]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+          registry-url: 'https://registry.npmjs.org'
+      - run: yarn install --frozen-lockfile --ignore-engines
+      - name: Canary Release
+        run: |
+          date=`date +%Y%m%d%H%M%S`
+          rm -f .changeset/*.md
+          project_version=$(cat package.json | jq '.version' | tr -d '"')
+          echo The project version is: $project_version        
+          [[ $project_version == 4* ]] && cp major-release-changeset.md .changeset          
+          [[ $project_version == 3* ]] && cp canary-release-changeset.md .changeset
```

Comment:
[req] If this workflow is only triggered targeting on main / v4 branch, why are we checking the project version? changesets for v4 should be independent from v3 right?

Reply:
I created the original workflow on the main branch and there we need this check (1. if we create a manual canary, 2. for future after we do a v4 release) so that we don't mistakenly release wrong versions.
This was copied here to allow for scheduled runs. We ideally dont need to have this check: [[ $project_version == 4* ]] as currently we're still at version 3 (even on main) and after the release (version will be 4 then), we'll switch the default branch anyway and this workflow will be disabled.